src/durableorchestrationcontext.ts

```diff
@@ -116,69 +104,21 @@ export class DurableOrchestrationContext {
      * Methods to handle collections of pending actions represented by [[Task]]
      * instances. For use in parallelization operations.
      */
-    public Task: ITaskMethods = {
-        all: (tasks: TaskBase[]) => {
-            let maxCompletionIndex: number | undefined;
-            const errors: Error[] = [];
-            const results: Array<unknown> = [];
-            for (const task of tasks) {
-                if (!TaskFilter.isCompletedTask(task)) {
-                    return TaskFactory.UncompletedTaskSet(tasks);
-                }
-
-                if (!maxCompletionIndex) {
-                    maxCompletionIndex = task.completionIndex;
-                } else if (maxCompletionIndex < task.completionIndex) {
-                    maxCompletionIndex = task.completionIndex;
-                }
-
-                if (TaskFilter.isFailedTask(task)) {
-                    errors.push(task.exception);
-                } else {
-                    results.push(task.result);
-                }
-            }
-
-            // We are guaranteed that maxCompletionIndex is not undefined, or
-            // we would have alreayd returned an uncompleted task set.
-            const completionIndex = maxCompletionIndex as number;
-
-            if (errors.length > 0) {
-                return TaskFactory.FailedTaskSet(
-                    tasks,
-                    completionIndex,
-                    new AggregatedError(errors)
-                );
-            } else {
-                return TaskFactory.SuccessfulTaskSet(tasks, completionIndex, results);
-            }
+    public Task = {
+        all: (tasks: Task[]): Task => {
+            const internalTasks = tasks.map((t) => t.internalTask);
+            const action = new WhenAllAction(internalTasks);
+            const innerTask = new WhenAllTask(internalTasks, action);
+            const task = new Task(innerTask);

```

Comment:
My intent with this encapsulation was to prevent users from accessing SDK-internal Task-methods in their orchestration code. However, your comment made me realize that I can effectively do that by giving end-users a constrained task `interface` instead. 

I've rewritten the code to do just that.